MoviesMVVM/MoviesMVVM/CoreDataService/CoreDataService.swift

```diff
@@ -0,0 +1,117 @@
+// CoreDataService.swift
+// Copyright © PozolotinaAA. All rights reserved.
+
+import CoreData
+import UIKit
+
+///  Загрузка данных с БД
+final class CoreDataService: CoreDataServiceProtocol {
+    // MARK: - Private Enum
+
+    private enum Constants {
+        static let movieObjectString = "MovieObject"
+        static let movieDetailObjectString = "MovieDetailObject"
+        static let emptyString = ""
+        static let failSaveString = "Failed to save movie: "
+        static let failRetrieveString = "Failed to retrieve movies:"
+    }
+
+    // MARK: - Private property
+
+    private let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
+
+    // MARK: - Public methods
+
+    func saveMovies(category: String, movies: [Movie]) {
+        context?.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
+        guard let context = context,
+              let entity = NSEntityDescription.entity(forEntityName: Constants.movieObjectString, in: context)
+        else { return }
+        do {
+            movies.forEach { movie in
+                let object = MovieObject(entity: entity, insertInto: context)
+                object.title = movie.title
+                object.id = Int64(movie.id)
+                object.overview = movie.overview
+                object.poster = movie.poster
+                object.rate = movie.rate
+                object.category = category
+            }
+            try context.save()
+        } catch {
+            print("\(Constants.failSaveString) \(error)")
```

Comment:
```suggestion
            print("\(Constants.failSaveString) \(error)")
```
Вроде ругаются на принты